platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/wiring/SignedStateReserver.java

```diff
@@ -0,0 +1,82 @@
+/*
+ * Copyright (C) 2023 Hedera Hashgraph, LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.swirlds.platform.wiring;
+
+import com.swirlds.platform.state.signed.ReservedSignedState;
+import edu.umd.cs.findbugs.annotations.NonNull;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Objects;
+import java.util.function.Consumer;
+
+/**
+ * Manages reservations of a signed state when it needs to be passed to one or more consumers. Whenever a state is
+ * passed to this instance, the assumption will be that it reserved for the reserver, and it is responsible for
```

Comment:
This description is a bit ambiguous - specifically "it reserved for the reserver" and "it is responsible", I'm not sure which subject the "it"s are referring to.

Reply:
I rewrote it completely since the interface changed